operations/dataGenV2.go

```diff
@@ -0,0 +1,332 @@
+		}
+	}
+	segmentWg.Wait()
+	Log.Debug.Printf("All Segments - Done !!!")
+	newUserWg.Wait()
+	Log.Debug.Printf("New Users - Done !!!")
+	Log.Debug.Printf("Main - Done !!!")
+}
+
+type SegmentProbMap struct {
+	EventCorrelationRangeMap map[string]utils.RangeMap
+	EventCorrelationMultiplier map[string]int
+	eventProbRangeMap utils.RangeMap
+	eventMultiplier int
+	activityProbRangeMap utils.RangeMap
+	activityMultiplier int
+}
+
+type ProbMap struct {
+	newUserRangeMap utils.RangeMap
+	newUserMultiplier int
+	segmentProbMap map[string]SegmentProbMap
+}
+
+func CreateNewUserProbMap()(utils.RangeMap, int){
+	
+	newUserProbablityMap := make(map[string]float64)
+	newUserProbablityMap["Insert"] = config.ConfigV2.New_user_probablity
+	newUserProbablityMap["NoInsert"] = (1.0 - config.ConfigV2.New_user_probablity)
+	return ComputeRangeMap(newUserProbablityMap)
+}
+
+func SeedUserOrNot(probMap ProbMap)bool{
+	r := rand.Intn(probMap.newUserMultiplier)
+	newUserInsert, state := probMap.newUserRangeMap.Get(r)
+	if(state == false){
+		Log.Error.Fatal(fmt.Sprintf("NewUserRangeMap: Key not found %v", r))
+	}
+	if(newUserInsert == "Insert") {
+		return true
+	}
+	return false
+}
+
+func GetRandomSegment()string{
+	segmentKeys := reflect.ValueOf(config.ConfigV2.User_segments).MapKeys()
+	seg := (segmentKeys[rand.Intn(len(segmentKeys))].Interface()).(string)
+	return seg
+}
+
+func IsAllSegmentsDone(segmentStatus map[string]bool) bool {
+
+	allSegmentsDone := true
+	for _,element := range segmentStatus {
+		if element == false {
+			allSegmentsDone = false
+			break
+		}
+	}
+	return allSegmentsDone
+}
+
+func OperateOnSegment(segmentWg *sync.WaitGroup, segmentName string, segment config.UserSegmentV2, probMap SegmentProbMap, userRangeStart int, userRangeEnd int, segmentStatus map[string]bool){
+
+	defer segmentWg.Done()
+	var wg sync.WaitGroup
+
+	Log.Debug.Printf("Main: Operating on %s with User Range %v - %v", segmentName , userRangeStart ,userRangeEnd)
+	//Generating events per user in the segment
+	for i := userRangeStart; i<= userRangeEnd; i++ {
+		wg.Add(1)
+		go GenerateEvents(
+			&wg,
+			segment,
+			(int)(config.ConfigV2.Activity_time_in_seconds / segment.Activity_ticker_in_seconds), 
+			config.ConfigV2.User_id_prefix+strconv.Itoa(i),
+			probMap)
+	}
+	
+	Log.Debug.Printf("Main: Waiting for %s to finish for user Range %v - %v", segmentName , userRangeStart , userRangeEnd)
+	wg.Wait()
+	segmentStatus[segmentName] = true
+	Log.Debug.Printf("Main: %s Completed for user Range %v - %v", segmentName, userRangeStart ,userRangeEnd)
+}
+
+func PreComputeRangeMap(segment config.UserSegmentV2) (SegmentProbMap) {
+
+	var probMap SegmentProbMap
+	probMap.EventCorrelationRangeMap = make(map[string]utils.RangeMap)
+	probMap.EventCorrelationMultiplier = make(map[string]int)
+	for item, element := range segment.Event_probablity_map.Correlation_matrix.Events {
+		probMap.EventCorrelationRangeMap[item], probMap.EventCorrelationMultiplier[item] = ComputeRangeMap(element)
```

Comment:
Better to manage these two variables in a struct and map the item to the struct.